src/codegen/Common.ts

```diff
@@ -36,6 +36,13 @@ const common = {
               .setDescription('Expression that will hide the page/form layout if true')
               .optional({ default: false }),
           ),
+          new CG.prop(
+            'expandedWidth',
+            new CG.bool()
+              .setTitle('Expanded width')
+              .setDescription('Expression that will expand the page/form layout if true')
+              .optional({ default: false }),
+          ),
```

Comment:
I added `expandedWidth` to `IPageSettings` in `settings.json` as an array that accepts the names of the pages that should be "expanded". Does this look like a better solution?

Should we gather settings like `order`, `excludeFromPdf`, and the new `expandedWidth` into a list of key-value pairs where each key points to an object containing each setting? Or do you think something like this could be done in another issue?

Eks. 1
```
    "pages": {
        "order": [
           "form",
            "summary",
            "grid"
        ],
        "excludeFromPdf": [
           "summary"
        ],
        "showLanguageSelector": true,
        "expandedWidth": ["grid"]
    },
```

Eks. 2
```
    "pages": {
        "settings": {
           "form": {
             "order": 0,
             "showLanguageSelector": true
           },
           "grid": {
             "order": 1,
             "expandedWidth": true,
             "showLanguageSelector": true
           },
           "summary": {
             "order": 2,
             "excludeFromPdf": true,
             "showLanguageSelector": true
           },
         }
    },
```

Reply:
> Or do you think something like this could be done in another issue?

In principle, yes, but at the same time it's harder to change this around later when we're adding new configuration options in this PR as well. So that's why I think it's best to clean it all up now that we have a nice chance (with v4).

That said, I'm not sure the suggestion where `expandedWidth` is an array containing page IDs is quite so smooth as I'd like. While the `order` and `excludeFromPdf` setting is made that way, it doesn't offer anything other than confusion if it's also possible to set directly in the layout file.

So, my thinking is, in the future we should have a tiered system where settings on pages can be set on multiple levels. Taking the `hideCloseButton` option as an example:
1. If set to `true` in `layout-sets.json` (you can add a `uiSettings` propety there, next to the `sets` property), the close button is hidden for all pages in all layout-sets. (This works in v4 already.)
2. If set to `true` in `App/ui/<set-name>/Settings.json`, in the allowed `pages.hideCloseButton` property, it is set for only that layout-set. If `true` in `layout-sets.json` but `false` in a specific `Settings.json`, that more specific `Settings.json` file should take precedence. (This also works in v4 now.)
3. (What I want to see happen:) If set to `true` in one specific `App/ui/<set-name>/layouts/<page-name>.json`, that again will override whatever is set in `Settings.json` and `layout-sets.json`

That way, making it an array of strings does not make much sense, as the setting is always boolean, and _where you set it_ decides for which pages it should count (while also making it possible for some pages to override the setting). I'd also like to move settings such as `excludeFromPdf` to that type of structure, but that's already out in the wild. (I'd even like to avoid double negatives such as `hideCloseButton = true` and turn it into a clearer `closeButton = false` or `showCloseButton = false`, but that's also an issue for later.)

So, no, I also don't particularly like to make to settings key-value pairs for each layout - if one layout needs specific settings we should configure that in the layout file itself, not in `Settings.json` for the whole layout-set.

Reply:
Aha, i forgot to mention I also removed the `expandedWidth` property from the layout-file. 
But if i understand it correctly i should add it back and then change the type of `expandedWidth` to also be `boolean` and then let the more specific setting override the other 😊 .

Reply:
Yup, if you add the property to `GlobalPageSettings` in `src/codegen/Common.ts` it should automatically be available in both `Settings.json` and `layout-sets.json`. It would also make sense to make it possible to configure `GlobalPageSettings` on each specific page, perhaps except from `autoSaveBehavior` as it would probably be a bit difficult to change the form data save behaviour per-page (at least right now). I guess that property could be moved out to a new type that extends `GlobalPageSettings` - so that it would only be possible to set that in `Settings.json` and `layout-sets.json`.